### ISSUE TITLE
dry-validation 1.0

### ISSUE THREAD

**Issue by Daniel-ltw** [michaelherold/interactor-contracts#19]
Is there any work to update to the latest dry-validation?

**Comment by michaelherold** [comment]
Nope, not yet. I would love this to work with 1.0+ so if you're interested, a PR would be welcome!

**Comment by StanBright** [comment]
@michaelherold based on your experience, do you think it will be a lot of work?

**Comment by vaihtovirta** [comment]
Hello all,
I want to prepare PR for this, because I've already started some work locally.

The minimal changes for making tests pass are relatively simple (considering some refactoring for later).
The only incompatibility issue I encountered is the inability to extend a schema once it's defined.
The reason behind that is class-based contracts introduced in `dry-validation#1.0` [raise DuplicateSchemaError](https://github.com/dry-rb/dry-validation/blob/6ce8f787f8375a3b2eef705c08298794dc3e055a/lib/dry/validation/contract/class_interface.rb#L203) if you try to add new rules to existing schema.

The issue could be solved by instantiating a new instance of `Dry::Validation::Contract` per each `expects` or `assures` block, and collecting reduced results from all these instances afterward.

Probably it's a question for you @michaelherold: I wonder what use-cases for having multiple blocks inside an interactor are? Is it still considered as desired behavior?
